#============================================
#  File: write_mirdeep2_mapper_output_summary_file.R
#  Directory Code:  /mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/scripts/
#  Date:  1/13/16
#  Description:  This code is used to create a summary of the reads mapped and unmapped for each library (n = 174)
#                Using sed on the command line to extract the correct lines from the OutputError file created after running
#                10_mirdeep2_genome_mapper.sh, then used this file in combination with the config_for_mapper.txt file to
#                create the summary file.
#============================================
#  Input File Directory:  /mnt/research/pigeqtl/analyses/microRNA/OutputsErrors/
#						  /mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/8_collapsed_fasta_output_expression_matrix/
#                         /mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/9_mirdeep2_genome_mapper_output/

#  Input File(s):   10_174_library_miRDeep2_genome_mapper.o29209455
#                   config_for_mapper.txt

#  Output File Directory: /mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/9_mirdeep2_genome_mapper_output/summary_files/

#  Output File(s): 174_library_mapper_outputerror_extraction.txt
#				   174_library_mapper_output_summary.txt
#============================================
#  Module used: sed (command line), R/3.1.0
#============================================

#First, on command line, the following commands were used to extract the correct lines from the output file generated by mapping the 174 libraries to the reference genome (file = 10_174_library_miRDeep2_genome_mapper.o29209455)
#cd /mnt/research/pigeqtl/analyses/microRNA/OutputsErrors/
#sed -n 1221,1396p 10_174_library_miRDeep2_genome_mapper.o29512834 > /mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/9_mirdeep2_genome_mapper_output/summary_files/174_library_mapper_outputerror_extraction.txt


#Then, in R:

#Set working directory:
setwd("/mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/9_mirdeep2_genome_mapper_output/summary_files")

mapstats<-read.table("174_library_mapper_outputerror_extraction.txt")
head(mapstats)
	#       V1        V2        V3       V4    V5    V6
	# 1 total: 232826977 174067218 58759759 0.748 0.252
	# 2   001:   2142071   1543592   598479 0.721 0.279
	# 3   002:   1210210    895792   314418 0.740 0.260
	# 4   003:   1393661   1065545   328116 0.765 0.235
	# 5   004:   1594434   1225437   368997 0.769 0.231
	# 6   005:   1933939   1334778   599161 0.690 0.310

#Reassign column names as they were in the original text file:
colnames(mapstats)<-c("id","total","mapped","unmapped","percentmapped","percentunmapped")
head(mapstats)
	#       id     total    mapped unmapped percentmapped percentunmapped
	# 1 total: 232826977 174067218 58759759         0.748           0.252
	# 2   001:   2142071   1543592   598479         0.721           0.279
	# 3   002:   1210210    895792   314418         0.740           0.260
	# 4   003:   1393661   1065545   328116         0.765           0.235
	# 5   004:   1594434   1225437   368997         0.769           0.231
	# 6   005:   1933939   1334778   599161         0.690           0.310

dim(mapstats)
	# [1] 175   6

#Remove "total" row:
mapstats<-mapstats[2:nrow(mapstats),]

head(mapstats)
	#     id   total  mapped unmapped percentmapped percentunmapped
	# 2 001: 2142071 1543592   598479         0.721           0.279
	# 3 002: 1210210  895792   314418         0.740           0.260
	# 4 003: 1393661 1065545   328116         0.765           0.235
	# 5 004: 1594434 1225437   368997         0.769           0.231
	# 6 005: 1933939 1334778   599161         0.690           0.310
	# 7 006: 1429073 1087694   341379         0.761           0.239


mapstats$id<-as.character(sub(":", "", mapstats$id))

dim(mapstats)
	# [1] 174   6

head(mapstats)
	#    id   total  mapped unmapped percentmapped percentunmapped
	# 2 001 2142071 1543592   598479         0.721           0.279
	# 3 002 1210210  895792   314418         0.740           0.260
	# 4 003 1393661 1065545   328116         0.765           0.235
	# 5 004 1594434 1225437   368997         0.769           0.231
	# 6 005 1933939 1334778   599161         0.690           0.310
	# 7 006 1429073 1087694   341379         0.761           0.239

#Now, pull in the config_for_mapper file to bring in the animal ids:
config<-read.table("/mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/8_collapsed_fasta_output_expression_matrix/config_for_mapper.txt")

head(config)
	#                V1 V2
	# 1 1034_express.fa  1
	# 2 1036_express.fa  2
	# 3 1041_express.fa  3
	# 4 1049_express.fa  4
	# 5 1058_express.fa  5
	# 6 1060_express.fa  6

#Convert the ids from config into characters, and maintain the 3-digit id:
config$V2<-as.character(sprintf("%03d",config$V2))

#Change column names:
colnames(config)<-c("inputfile", "id")
head(config)
	#            animal  id
	# 1 1034_express.fa 001
	# 2 1036_express.fa 002
	# 3 1041_express.fa 003
	# 4 1049_express.fa 004
	# 5 1058_express.fa 005
	# 6 1060_express.fa 006

#Use "merge" function to bring these two datasets together, mergeing by "id" column:
mapsummary<-merge(config, mapstats, by = "id")
head(mapsummary)

	#    id          animal   total  mapped unmapped percentmapped percentunmapped
	# 1 001 1034_express.fa 2142071 1543592   598479         0.721           0.279
	# 2 002 1036_express.fa 1210210  895792   314418         0.740           0.260
	# 3 003 1041_express.fa 1393661 1065545   328116         0.765           0.235
	# 4 004 1049_express.fa 1594434 1225437   368997         0.769           0.231
	# 5 005 1058_express.fa 1933939 1334778   599161         0.690           0.310
	# 6 006 1060_express.fa 1429073 1087694   341379         0.761           0.239

#Check that no data was shuffled around in the merge:
sum(mapsummary[,c(1,3:7)] == mapstats)
	# [1] 1044
dim(mapstats)
	# [1] 174   6
174*6
	# [1] 1044
#Looks good!

#Calculate the average mapped reads:
sum(mapsummary$mapped)
	# [1] 174067218
sum(mapsummary$total)
	# [1] 232826977
(sum(mapsummary$mapped))/(sum(mapsummary$total))
	# [1] 0.7476248

mean(mapsummary$total)
# [1] 1338086
median(mapsummary$total)
# [1] 1288872
range(mapsummary$total)
# [1]  571643 2513305
mean(mapsummary$mapped)
# [1] 1000386
median(mapsummary$mapped)
# [1] 973972
range(mapsummary$mapped)
# [1]  326303 1834999

summary(mapsummary$total)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
# 571600 1124000 1289000 1338000 1571000 2513000

summary(mapsummary$mapped)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
# 326300  848800  974000 1000000 1155000 1835000

summary(mapsummary$percentmapped)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max.
# 0.5110  0.7470  0.7640  0.7509  0.7740  0.8020

#Total reads mapped = 74.76%

#Save file:
write.table(mapsummary, file = "/mnt/research/pigeqtl/analyses/microRNA/1_preprocess_fastq_files/9_mirdeep2_genome_mapper_output/summary_files/174_library_mapper_output_summary.txt", quote = FALSE, sep = " ", row.names = FALSE, col.names = TRUE)







